pkg/provisioner/ironic/delete_test.go

```diff
@@ -0,0 +1,192 @@
+package ironic
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/gophercloud/gophercloud/openstack/baremetal/v1/nodes"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/metal3-io/baremetal-operator/pkg/bmc"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/clients"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/testserver"
+)
+
+func TestDelete(t *testing.T) {
+
+	nodeUUID := "33ce8659-7400-4c68-9535-d10766f07a58"
+
+	cases := []struct {
+		name      string
+		ironic    *testserver.IronicMock
+		inspector *testserver.InspectorMock
+		hostName  string
+
+		expectedDirty        bool
+		expectedRequestAfter int
+		expectedUpdate       *nodes.UpdateOperation
+
+		expectedError string
+	}{
+		{
+			name: "delete-host-fail",
+			ironic: testserver.NewIronic(t).Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+				},
+			).DeleteError(nodeUUID, http.StatusInternalServerError),
+			expectedError: "failed to remove host",
+		},
+		{
+			name: "delete-host-busy",
+			ironic: testserver.NewIronic(t).Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+				},
+			).DeleteError(nodeUUID, http.StatusConflict),
+			expectedDirty:        true,
+			expectedRequestAfter: 0,
+		},
+		{
+			name: "delete-host-not-found",
+			ironic: testserver.NewIronic(t).Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+				},
+			).DeleteError(nodeUUID, http.StatusNotFound),
+			expectedDirty:        true,
+			expectedRequestAfter: 0,
+		},
+		{
+			name: "delete-ok",
+			ironic: testserver.NewIronic(t).Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+				},
+			).Delete(nodeUUID),
+			expectedDirty:        true,
+			expectedRequestAfter: 0,
+		},
+		{
+			name: "host-not-found",
+
+			hostName: "worker-0",
+			ironic:   testserver.NewIronic(t).Ready().NodeError(nodeUUID, http.StatusGatewayTimeout),
+
+			expectedError: "failed to find existing host: failed to find node by ID 33ce8659-7400-4c68-9535-d10766f07a58: Expected HTTP response code \\[200\\].*",
+		},
+		{
+			name:   "not-ironic-node",
+			ironic: testserver.NewIronic(t).Ready().NoNode(nodeUUID).NoNode("myhost"),
+		},
+		{
+			name: "available-node",
+			ironic: testserver.NewIronic(t).WithDefaultResponses().Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "available",
+				},
+			),
+			expectedDirty:        true,
+			expectedRequestAfter: 10,
```

Comment:
Should we refer to the variable that controls that timeout, instead of using a hard-coded value? That way if we adjust the timeout we won't have to modify the tests.

Reply:
Fixed